src/views/common/Animation/AnimationStatePlaybackBarLayer.tsx

```diff
@@ -1,175 +1,156 @@
+        dragSelectInitiationRef.current = window.setTimeout(() => {
+            dragSelectInitiationRef.current = undefined
+            initiateDragSelection(x)
+        }, dragSelectInitiationDelayMs)
+    }, [getEventPoint, isPlaying, initiateScrubbing, initiateDragSelection])
+    
     const handleMouseUp = useCallback((e: React.MouseEvent) => {
-        if (draggingXRef.current && wasPlayingRef.current && !isPlaying) {
-            dispatch({type: 'TOGGLE_PLAYBACK'})
-            wasPlayingRef.current = false
+        if (dragSelectInitiationRef.current !== undefined) {
+            // mouseup within dragSelectInitiationDelayMs after mousedown === timedout callback hasn't yet fired.
+            // This corresponds to an actual click, not a drag initiation, so do the click handling stuff
+            clearTimeout(dragSelectInitiationRef.current)
+            dragSelectInitiationRef.current = undefined
+            const [x, y] = getEventPoint(e)
+            handleScrollbarClick(x, y)
+            updateSelectedWindow(undefined) // TODO: Check for possible race condition or double-rerender?
+            return
         }
-        draggingXRef.current = undefined
-    }, [dispatch, isPlaying])
-
-    // TODO: Break drag debounce stuff into a more self-contained hook?
-    const dragTargetFrameRef = useRef<number | undefined>(undefined)
-    const nextDebounceRequest = useRef<number | undefined>(undefined)
-    const debounceSetFrame = useCallback(() => {
-        if (draggingXRef.current && dragTargetFrameRef.current) {
-            if (dragTargetFrameRef.current !== currentFrameIndex) {
-                dispatch({type: 'SET_CURRENT_FRAME', newIndex: dragTargetFrameRef.current})
-            }
-            nextDebounceRequest.current = window.requestAnimationFrame(debounceSetFrame)
-        } else {
-            if (nextDebounceRequest.current) {
-                window.cancelAnimationFrame(nextDebounceRequest.current)
-            }
-            nextDebounceRequest.current = undefined
-            dragTargetFrameRef.current = undefined
-        }
-    }, [currentFrameIndex, dispatch])
-
+        terminateScrubbing(isPlaying)
+        terminateDragSelection()
+    }, [getEventPoint, terminateScrubbing, terminateDragSelection, handleScrollbarClick, updateSelectedWindow, isPlaying])
+    
     const handleMouseMove = useCallback((e: React.MouseEvent) => {
-        if (!draggingXRef.current) return
-
-        const [x, y] = getEventPoint(e)
-        if (Math.abs(x - draggingXRef.current) > 2) {
-            dragTargetFrameRef.current = getNewFrame(x, y)
-            draggingXRef.current = x
-            debounceSetFrame()
+        if (e.buttons === 0) { // terminate drags where mouseup happened outside our element
+            terminateScrubbing(isPlaying)
+            terminateDragSelection()
+            return // short-circuit processing of no-button mouse moves--we don't care about them
```

Comment:
This can happen if the user clicks, drags, then releases the mouse button outside the frame. Then no mouseup event ever gets generated, so we keep scrubbing/dragging even though the button isn't held down.